components/TxForm.tsx

```diff
@@ -1,35 +1,54 @@
-import React, { useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { Button } from '@chakra-ui/button';
-import { FormControl, FormLabel } from '@chakra-ui/form-control';
-import { Input, InputGroup, InputRightElement } from '@chakra-ui/input';
-import { Box, Flex } from '@chakra-ui/layout';
+import { FormControl, FormErrorMessage, FormLabel } from '@chakra-ui/form-control';
+import { Input, InputGroup } from '@chakra-ui/input';
+import { Box } from '@chakra-ui/layout';
 import { useDisclosure } from '@chakra-ui/react';
 import { getRailgunSmartWalletContractForNetwork } from '@railgun-community/quickstart';
+import { validateRailgunAddress } from '@railgun-community/quickstart';
 import { erc20ABI } from '@wagmi/core';
+import { GetNetworkResult, watchNetwork } from '@wagmi/core';
 import { BigNumber, constants, ethers } from 'ethers';
+import { parseUnits } from 'ethers/lib/utils.js';
 import { useContractWrite, useNetwork, usePrepareContractWrite } from 'wagmi';
 import ReviewTransactionModal from '@/components/ReviewTransactionModal';
 import TokenInput from '@/components/TokenInput';
 import { useToken } from '@/contexts/TokenContext';
+import { TokenListContextItem } from '@/contexts/TokenContext';
 import useNotifications from '@/hooks/useNotifications';
 import useTokenAllowance from '@/hooks/useTokenAllowance';
 import { ethAddress } from '@/utils/constants';
-import { networks } from '@/utils/networks';
+import { buildBaseToken, networks } from '@/utils/networks';
 
-export const TxForm = () => {
-  // TODO: Placeholder notification for shielding
-  const { tokenAllowances } = useToken();
+type TxFormValues = {
+  recipient: string;
+  amount: string;
+  token: string;
+};
+
+export const TxForm = ({ recipientAddress }: { recipientAddress?: string }) => {
+  const { tokenAllowances, tokenList } = useToken();
   const { chain } = useNetwork();
   const network = networks[chain?.id || 1];
-  const { notifyUser, txNotify } = useNotifications();
+  const { notifyUser } = useNotifications();
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    reset,
+    formState: { errors },
+  } = useForm<TxFormValues>({
+    defaultValues: {
+      token: network.baseToken.name,
+      recipient: recipientAddress,
+    },
+  });
   const { isOpen: isReviewOpen, onOpen: onReviewOpen, onClose: onReviewClose } = useDisclosure();
```

Comment:
nit (due to time, would probably ignore)

```suggestion
  const { isOpen: isReviewOpen, onOpen: openReview, onClose: closeReview } = useDisclosure();
```